File: Python/f/f6-Area.py
'''
6. Calculo de area de figuras geométricas.
'''

import math

while True:
    print("1. Triangulo\n2. Cuadrado\n3. Circulo\n")
    op = int(input("Ingrese la figura que desea calcular: "))
    match op:
        case 1:
            base = int(input("Ingrese la base del triangulo: "))
            altura = int(input("Ingrese la altura del triangulo: "))
            print("El area del triangulo es: " + str(base*altura*0.5))
            break
        case 2:
            lado = int(input("Ingrese el lado del cuadrado: "))
            print("El area del cuadrado es: " + str(lado*lado))
            break
        case 3:
            radio = int(input("Ingrese el radio del circulo: "))
            print("El area del circulo es: " + str(math.pi*radio*radio))
            break
        case _:
            break

File: Rust/f/f6-Area.rs
// 6. Calculo de area de figuras geométricas.
use std::io;
fn input() -> i32 {
    loop{
    let mut input = String::new();
    let _ = io::stdin().read_line(&mut input);
    match input.trim().parse::<i32>() {
        Ok(n) => break n,
        Err( _ ) => {
            println!("Ingrese un número válido: ");
        },
    }
    }
}

fn inputf() -> f32 {
    loop{
    let mut input = String::new();
    let _ = io::stdin().read_line(&mut input);
    match input.trim().parse::<f32>() {
        Ok(n) => break n,
        Err( _ ) => {
            println!("Ingrese un número válido: ");
        },
    }
    }
}

fn main() {
    println!("1.Triangulo");
    println!("2.Cuadrado");
    println!("3.Circulo");
    println!("Ingresa la operacion: ");
    let op:i32 = input();
    match op {
        1 => {
            println!("Ingresa la base: ");
            let b:f32 = inputf();
            println!("Ingresa la altura: ");
            let h:f32 = inputf();
            println!("El area es: {}", b*h/2.0);
        },
        2 => {
            println!("Ingresa el lado: ");
            let l:f32 = inputf();
            println!("El area es: {}", l*l);
        },
        3 => {
            println!("Ingresa el radio: ");
            let r:f32 = inputf();
            println!("El area es: {}", 3.14*r*r);

        },
        _ => {

        }

    }

}
